Lib/idlelib/pyshell.py

```diff
@@ -726,6 +726,17 @@ def showsyntaxerror(self, filename=None):
         msg = getattr(value, 'msg', '') or value or "<no detail available>"
         lineno = getattr(value, 'lineno', '') or 1
         offset = getattr(value, 'offset', '') or 0
+
+        # special case to fix bpo-43600
+        if "f-string:" in msg:
```

Comment:
Using `in` is a little fragile, since one could do `raise SyntaxError("just some text containing f-string:")`. That's why the issue recommend testing if msg starts with

I also think the offset math isn't quite right for triple quotes

Reply:
Fixed